src/shogun/kernel/SpectralMixtureKernel.h

```diff
@@ -0,0 +1,185 @@
+/*
+ * This software is distributed under BSD Clause 3 license (see LICENSE file).
+ *
+ * Written (W) 2015 Esben Soerig
+ */
+
+#ifndef SPECTRALMIXTUREKERNEL_H
+#define SPECTRALMIXTUREKERNEL_H
+
+#include <shogun/kernel/DotKernel.h>
+
+namespace shogun
+{
+	class CDotFeatures;
+
+/** @brief The spectral mixture kernel
+ * 
+ * TODO
+ *
+ */
+
+class CSpectralMixtureKernel: public CDotKernel
+{
+	public:
+		/** default constructor */
+		CSpectralMixtureKernel();
+
+		/** constructor
+		 *
+		 * The num_components argument specifies how many spectral mixture 
+		 * components to initialize. The gaussian spectral mixture components are
+		 * initilized stochastically using a standard initialization scheme  
+		 * based on metrics from the left-hand and right-hand side features.
+		 *
+		 * @param l features of left-hand side
+		 * @param r features of right-hand side
+		 * @param num_components number of spectral mixture components
+		 * @param size cache size. Default value: 10
+		 */
+		CSpectralMixtureKernel(CDotFeatures* l, CDotFeatures* r,
+			int32_t num_components, int32_t size=10);
+
+		/** constructor with specific parameter initializations.
+		*
+		* The number of mixture components is implicitly specified by the
+		* lengths of the weights, means, and stddeviations vectors (which must 
+		* be of equal length).
+		*
+		* @param weights vector of weights for each mixture component
+		* @param means vector of means for each mixture component
+		* @param stddeviations vector of standard deviations for each mixture component
+		* @param size cache size. Default value: 10
+		*/
+		CSpectralMixtureKernel(const SGVector<float64_t>& weights,
```

Comment:
Brevity :) 

Not sure about any benefits. SGVectors are usually passed by value